src/JET.jl

```diff
@@ -421,7 +421,15 @@ const StateAtPC = Tuple{State,Int}
 const LineTable = Union{Vector{Any},Vector{LineInfoNode}}
 
 get_stmt((sv, pc)::StateAtPC)         = @inbounds sv.src.code[pc]
-get_lin((sv, pc)::StateAtPC)          = @inbounds (sv.src.linetable::LineTable)[sv.src.codelocs[pc]]::LineInfoNode
+get_lin((sv, pc)::StateAtPC)          = begin
+    if 1 <= sv.src.codelocs[pc] <= length(sv.src.codelocs)
```

Comment:
Can this check replaced with something like `length(codelocs) == 0 || length(linetable) == 0`?
I assume Zygote may have empty `codelocs`/`linetable` when the source is unknown, but otherwise (if it can have arbitrary length vectors) there is no reason to trust their location information.

Reply:
Both `codelocs` and `linetable` are defined in the test case, but `codelocs[pc]` contains the invalid index 0.

Reply:
I see. `length(sv.src.codelocs)` still should be `length(sv.src.linetable)` though?

Reply:
Ouch...